src/main/java/xyz/nucleoid/plasmid/command/FriendCommand.java

```diff
@@ -0,0 +1,122 @@
+package xyz.nucleoid.plasmid.command;
+
+import com.mojang.authlib.GameProfile;
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import net.minecraft.command.argument.GameProfileArgumentType;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.LiteralText;
+import net.minecraft.util.UserCache;
+import xyz.nucleoid.plasmid.command.Friend.FriendList;
+import xyz.nucleoid.plasmid.command.Friend.FriendListManager;
+
+import java.util.UUID;
+
+import static net.minecraft.server.command.CommandManager.argument;
+import static net.minecraft.server.command.CommandManager.literal;
+
+
+/*
+    some breaking problems are if a user cache name expires, the whole thing breaks
+    at the moment i have no idea how to fix this besides saving player names ourselves
+ */
+
+public class FriendCommand {
+    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
+        dispatcher.register(literal("friend")
+                .then(literal("accept").then(argument("player", GameProfileArgumentType.gameProfile()).executes(FriendCommand::acceptFriend)))
+                .then(literal("add").then(argument("playerName", GameProfileArgumentType.gameProfile()).executes(FriendCommand::registerFriendAdd)))
+                .then(literal("list").executes(FriendCommand::ListFriends))
+                .then(literal("requests").executes(FriendCommand::ListRequests)));
+    }
+
+    private static int ListRequests(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
```

Comment:
```suggestion
    private static int executeListRequests(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
```

no PascalCase in Java methods.